runtime/rasdump/javadump.cpp

```diff
@@ -1484,6 +1484,94 @@ JavaCoreDumpWriter::writeEnvUserArgsHelper(J9VMInitArgs *vmArgs)
 				_OutputStream.writeCharacters("\n");
 			}
 		}
+
+		/* Ignored options for -Xjit and -Xaot follows these rules:
+		 * 1. If -Xjit is followed by -Xint or -Xnojit,
+		 *    then any -Xjit:optionString beforehand appears as ignored.
+		 * 2. If XX:+MergeCompilerOptions is present,
+		 *    then all -Xjit options after -Xint or -Xnojit should not be ignored.
+		 * 3. If multiple -Xjit strings appear, all but the last one will appear as ignored.
+		 *
+		 * Similar rules apply for -Xaot.
+		 */
+
+		jint numXjit = 0;
+		jint numXaot = 0;
+		bool hasXXMerge = false;
+
+		/* These indices ensure all the options before them will be printed as ignored. */
+		jint ignoreXjitBefore = 0;
+		jint ignoreXaotBefore = 0;
+
+		for (jint i = 0; i < args->nOptions; i++) {
+			const char *optionString = args->options[i].optionString;
+
+			if (0 == strncmp(optionString, "-Xint", 6)) {
+				/* Case 1: ignores all options before. */
+				ignoreXjitBefore = numXjit;
+				ignoreXaotBefore = numXaot;
+			} else if (0 == strncmp(optionString, "-Xnojit", 8)) {
+				/* Case 1: ignore all jit options before. */
+				ignoreXjitBefore = numXjit;
+			} else if (0 == strncmp(optionString, "-Xnoaot", 8)) {
+				/* Case 1: ignore all aot options before. */
+				ignoreXaotBefore = numXaot;
+			} else if (0 == strncmp(optionString, "-XX:+MergeCompilerOptions", 26)) {
+				/* Case 2 */
+				hasXXMerge = true;
+			} else if (0 == strncmp(optionString, "-XX:-MergeCompilerOptions", 26)) {
+				/* If -XX:-MergeCompilerOptions is encountered after
+				 * -XX:+MergeCompilerOptions, the former is overriden. */
+				hasXXMerge = false;
+			} else if (0 == strncmp(optionString, "-Xjit:", 6)) {
+				/* Case 3 */
+				numXjit += 1;
+			} else if (0 == strncmp(optionString, "-Xaot:", 6)) {
+				/* Case 3 */
+				numXaot += 1;
+			}
+		}
+
+		int countXjit = 0;
+		int countXaot = 0;
```

Comment:
The type of these variables should be `jint`, not `int`. They should be incremented with `+= 1`.